### ISSUE TITLE
Installer and uninstaller crash on exit

### ISSUE THREAD

**Issue by Radrik5** [git-for-windows/git#657]
Version: Git for Windows 2.7.1(2). Previous version (Git for Windows 2.7.1) works fine.

Installation options:
```
Path Option: BashOnly
SSH Option: OpenSSH
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Disabled
```

Exception in installer after I press "Finish":
`Unhandled exception at 0x3AA645DA in Git-2.7.1.2-64-bit.tmp: 0xC0000005: Access violation executing location 0x3AA645DA.`

Exceptions in uninstaller after uninstallation is complete:
`Unhandled exception at 0x3AA645DA in unins000.exe: 0xC0000005: Access violation executing location 0x3AA645DA.`
`Unhandled exception at 0x3AA645DA in _iu14D2N.tmp: 0xC0000005: Access violation executing location 0x3AA645DA.`

Crash dumps:
[Git-2.7.1.2-64-bit.dmp.7z](https://drive.google.com/file/d/0B6gywcQjjFvDNF9ORC02V0R0WHc/view?usp=sharing)
[unins000.dmp.7z](https://drive.google.com/file/d/0B6gywcQjjFvDbEE1Uk1UdEJKMlk/view?usp=sharing)
[_iu14D2N.dmp.7z](https://drive.google.com/file/d/0B6gywcQjjFvDUEllbXBteWtaOVE/view?usp=sharing)

**Comment by dscho** [comment]
> ```
> Git-2.7.1.2-64-bit.dmp.7z is unsafe to download and was blocked by SmartScreen Filter.
> ```

Hmm.

**Comment by dscho** [comment]
Apart from this problem, there is another, more serious one: where is the information about the OS, the information whether this is run via admin, power user or plain user, the install log (that you get when calling the installer with the `/LOG=<filename>` option)?

**Comment by Radrik5** [comment]
Sorry, I didn't know about this option. Log file is attached: [Git-2.7.1.2-64-bit.exe.txt](https://github.com/git-for-windows/git/files/132118/Git-2.7.1.2-64-bit.exe.txt)

Extract from the log file:
```
Windows version: 6.1.7601 SP1  (NT platform: Yes)
User privileges: Administrative
```

> Git-2.7.1.2-64-bit.dmp.7z is unsafe to download and was blocked by SmartScreen Filter.

SmartScreen Filter is wrong. It's just a 7zip archive with a dump file. I could attach unpacked dump file but it 10 times larger that the archive.

Stack trace with the exception from Git-2.7.1.2-64-bit.dmp:
```
0:000> k
ChildEBP RetAddr  
WARNING: Frame IP not in any known module. Following frames may be wrong.
0018fe6c 77049441 <Unloaded_LvHook.dll>+0x645da
0018fe88 77069cfa ntdll!RtlProcessFlsData+0x57
0018ff20 77069c46 ntdll!LdrShutdownProcess+0xbd
0018ff34 76b179dc ntdll!RtlExitUserProcess+0x74
0018ff48 00405197 kernel32!ExitProcessStub+0x12
0018ff60 004051ba Git_2_7_1_2_64_bit+0x5197
0018ff64 00500765 Git_2_7_1_2_64_bit+0x51ba
0018ff88 76b1337a Git_2_7_1_2_64_bit+0x100765
0018ff94 77049882 kernel32!BaseThreadInitThunk+0xe
0018ffd4 77049855 ntdll!__RtlUserThreadStart+0x70
0018ffec 00000000 ntdll!_RtlUserThreadStart+0x1b
```

LvHook.dll is a part of ABBYY Lingvo x5 15.0.511.0 and indeed when I exit the Lingvo application, Git for Windows 2.7.1(2) installer and uninstaller stop crashing. When I start Lingvo, the installer and uninstaller start crashing again. I haven't seen such problems before with other applications and all the previous versions of Git for Windows worked fine.

**Comment by Radrik5** [comment]
ABBYY Lingvo is a dictionary, Russian-English in my case. LvHook.dll is needed to support hotkeys in other applications, e.g. Ctrl+C+C opens Lingvo with translation of the selected word.

**Comment by dscho** [comment]
I doubt that Git's installer is the first application to cause problems with LvHook. Have you launched a web search?

**Comment by Radrik5** [comment]
Well, yes, it's not the first application in history that has problems with LvHook.dll but it's the first application on my PC and the first Git's installer that has problems with LvHook.dll. What changed in 2.7.1(2) installer that forces it unload LvHook.dll before unregistering its hook or changes load address of LvHook.dll?

**Comment by dscho** [comment]
> What changed in 2.7.1(2) installer

It loads some system `.dll` files explicitly now. And of course, they are unloaded at exit. So it looks as if a bug in `LvHook.dll` is now uncovered.

I strongly tend toward closing this ticket as *wontfix*...

**Comment by dscho** [comment]
> I strongly tend toward closing this ticket as wontfix...

And now I actually do it.
